sqlx-mysql/src/connection/infile.rs

```diff
@@ -0,0 +1,298 @@
+//! Support for `LOAD DATA LOCAL INFILE` statements
+//!
+//! This MySQL feature allows the client to send a local file to the server, which is then
+//! loaded into a table. This should be faster than sending the data row-by-row.
+//!
+//! # Example
+//! ```rust,no_run
+//! use sqlx::mysql::infile::MySqlPoolInfileExt;
+//!
+//! #[tokio::main]
+//! async fn main() -> Result<(), sqlx::Error> {
+//!     let pool = sqlx::mysql::MySqlPool::connect("mysql://root:password@localhost:3306/sqlx").await?;
+//!
+//!     let res = {
+//!         let mut stream = pool
+//!             .load_local_infile("LOAD DATA LOCAL INFILE 'dummy' INTO TABLE testje")
+//!             .await?;
+//!         stream.send(b"1\n2\n3\n4\n5\n6\n7\n8\n9\n10").await?;
+//!         stream.finish().await?
+//!     };
+//!     println!("{}", res); // 10
+//!
+//!     Ok(())
+//! }
+//! ```
+
+use std::{
+    ops::{Deref, DerefMut},
+    pin::Pin,
+    task::Poll,
+};
+
+use crate::error::Error;
+use crate::protocol::response::LocalInfilePacket;
+use crate::protocol::text::Query;
+use crate::sqlx_core::net::Socket;
+use futures_core::{future::BoxFuture, ready};
+use futures_io::AsyncWrite;
+use sqlx_core::pool::{Pool, PoolConnection};
+
+use crate::{MySql, MySqlConnection};
+
+use super::MySqlStream;
+
+/// Extension of the MySql Pool with support for `LOAD DATA LOCAL INFILE` statements.
+pub trait MySqlPoolInfileExt {
+    /// Execute the query using the given handler.
+    ///
+    /// See the [infile](`crate::infile`) module documentation for an example.
```

Comment:
```suggestion
    /// Check out a connection and execute a `LOAD DATA LOCAL INFILE` query.
    ///
    /// This will return an error if `statement` is not a `LOAD DATA LOCAL INFILE` query.
    ///
    /// Refer to the [MySQL documentation] or [MariaDB documentation] for details
    /// on the syntax of this query.
    ///
    /// The filename given in the query is not important, as this API does not touch the filesystem,
    /// but is available for reference as [`MySqlLocalInfile::filename()`] if it is relevant to your application.
    ///
    /// See the [infile](`crate::infile`) module documentation for an example.
    ///
    /// [MySQL documentation]: https://dev.mysql.com/doc/refman/8.0/en/load-data.html
    /// [MariaDB documentation]: https://mariadb.com/kb/en/load-data-infile/
```